crates/goose/src/providers/openrouter.rs

```diff
@@ -211,6 +234,27 @@ impl Provider for OpenRouterProvider {
         self.model.clone()
     }
 
+    async fn structure_response(
```

Comment:
Would it make sense to make this (or a more generic version of this) the default implementation? That way you don't have to touch the agents?